### ISSUE TITLE
Fix cycles UI for hailstoneenglish.s

### ISSUE THREAD

**Issue by lupino3** [lupino3/edumips64#134]
Running the hailstoneenglish.s test, the Cycles window becomes garbled after some time.

This is probably related to the warnings seen during tests:

```
[junit] Aug 23, 2016 1:26:34 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State WAW is not allowed after state RAW
[junit] Aug 23, 2016 1:26:34 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State WAW is not allowed after state RAW
[junit] Aug 23, 2016 1:26:34 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State WAW is not allowed after state RAW
[junit] Aug 23, 2016 1:26:35 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State WB is not allowed after state IF
[junit] Aug 23, 2016 1:26:35 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State ID is not allowed after state WB
[junit] Aug 23, 2016 1:26:35 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State WB is not allowed after state IF
[junit] Aug 23, 2016 1:26:35 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State ID is not allowed after state WB
[junit] Aug 23, 2016 1:26:35 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State WB is not allowed after state IF
[junit] Aug 23, 2016 1:26:35 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State ID is not allowed after state WB
[junit] Aug 23, 2016 1:26:35 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State WB is not allowed after state IF
[junit] Aug 23, 2016 1:26:35 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State ID is not allowed after state WB
[junit] Aug 23, 2016 1:26:35 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State WB is not allowed after state IF
[junit] Aug 23, 2016 1:26:35 PM org.edumips64.ui.common.CycleElement addState
[junit] SEVERE: State ID is not allowed after state WB
```

**Comment by lupino3** [comment]
It seems that there are 2 classes of errors here.

The first one is about `WAW` after `RAW`, which is due to one of the FPU tests. I think that one is legit and I will verify it.

All the other errors seem to not appear always: I ran the tests twice and they appeared only in one run.

**Comment by lupino3** [comment]
The error about `WAW` after `RAW` is fixed. I am working on #139, which might be related.

**Comment by lupino3** [comment]
The UI becoming garbled is not related to #139 or to any of those error logs. I have a fix for #139 which I am about to push, but the UI still becomes garbage after some time; in my test, this happens around cycle 4000.

**Comment by lupino3** [comment]
It happens precisely at cycle 4370. The interesting thing is that at cycle 4369 instruction in cycle 1 is shown again. I reproduced this in a test build that also prints the cycle number to the left of the instruction:

![wrong](https://cloud.githubusercontent.com/assets/200139/26283034/53e8f1de-3e17-11e7-9f45-63efd2c96682.png)

Looks like some sort of overflow..
